### PR TITLE
Synchronize with Azure/azure-cli upstream

### PR DESCRIPTION
**Pull request by troy0820**
#### Synchronize Azure/azure-cli `aro` to ARO-RP 

- [x]  python]Add examples to help
- [x] Add function to validate int for vm count and size _(from upstream)_
- [x] Sychronize with upstream azure/azure-cli

PR to fix function with the wrong flag in the error message to keep in sync 
- https://github.com/Azure/azure-cli/pull/13439

### CODE REVIEW

File: python/az/aro/azext_aro/_validators.py [hunk 1]
```diff
@@ -70,6 +70,15 @@ def validate_domain(namespace):
                            namespace.domain)
 
 
+def _validate_int(key, i):
```
Comment (jim-minter) [inline python/az/aro/azext_aro/_validators.py:73]: this is re-adding something that should be being removed.

File: python/az/aro/azext_aro/_help.py [hunk 2]
```diff
@@ -12,31 +12,53 @@
 helps['aro create'] = """
     type: command
     short-summary: Create a cluster.
+    examples:
+      - name: Create an OpenShift cluster
```
Comment (jim-minter) [inline python/az/aro/azext_aro/_help.py:16]:
```suggestion
      - name: Create an OpenShift cluster.
```

All of these should end with `.`.

File: python/az/aro/azext_aro/_help.py [hunk 3]
```diff
@@ -12,31 +12,53 @@
...
+        text: az aro create --resource-group MyResourceGroup --name MyCluster  --vnet MyVnet --master-subnet MyMasterSubnet --worker-subnet MyWorkerSubnet
+      - name: Create an OpenShift cluster with 5 compute nodes and Red Hat Pull Secret
```
Comment (jim-minter) [inline python/az/aro/azext_aro/_help.py:18]:
```suggestion
      - name: Create an OpenShift cluster with 5 compute nodes and Red Hat pull secret.
```

File: python/az/aro/azext_aro/_validators.py (hunk 1 above)
Comment (troy0820) [reply python/az/aro/azext_aro/_validators.py:73]: I can remove this.  This was added to the azure/azure-cli repository.

Comment (jim-minter) [reply python/az/aro/azext_aro/_validators.py]: I think it was removed here with the intention of updating azure/azure-cli and removing it there too.

File: python/az/aro/azext_aro/_help.py [hunk 4]
```diff
@@ -12,31 +12,53 @@
 helps['aro create'] = """
     type: command
     short-summary: Create a cluster.
+    examples:
+      - name: Create an OpenShift cluster.
+        text: az aro create --resource-group MyResourceGroup --name MyCluster  --vnet MyVnet --master-subnet MyMasterSubnet --worker-subnet MyWorkerSubnet
```
Comment (jim-minter) [inline python/az/aro/azext_aro/_help.py:17]:
```suggestion
        text: az aro create --resource-group MyResourceGroup --name MyCluster --vnet MyVnet --master-subnet MyMasterSubnet --worker-subnet MyWorkerSubnet
```

**Pull request merged**